airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/S3Destination.java

```diff
@@ -60,6 +71,12 @@ public AirbyteConnectionStatus check(final JsonNode config) {
     }
   }
 
+  public static void testIAMUserHasListObjectPermission(final AmazonS3 s3Client, final String bucketName) {
+    LOGGER.info("Started testing if IAM user can call listObjects on the destination bucket");
+    s3Client.listObjects(bucketName);
```

Comment:
Is there a limit param we can add? The bucket may have a ton of objects

Reply:
> Because buckets can contain a virtually unlimited number of keys, the complete results of a list query can be extremely large. To manage large result sets, Amazon S3 uses pagination to split them into multiple responses. Always check the ObjectListing.isTruncated() method to see if the returned listing is complete or if additional calls are needed to get more results. Alternatively, use the AmazonS3Client.listNextBatchOfObjects(ObjectListing) method as an easy way to get the next page of object listings.

maybe worth to look into `listObjectsV2` ? https://stackoverflow.com/a/37539994

Reply:
I added `ListObjectsRequest` with limit on number of keys to return